src/daft-core/src/array/ops/from_arrow.rs

```diff
@@ -64,21 +70,61 @@ impl FromArrow for FixedSizeListArray {
 impl FromArrow for ListArray {
     fn from_arrow(field: FieldRef, arrow_arr: Box<dyn arrow2::array::Array>) -> DaftResult<Self> {
         match (&field.dtype, arrow_arr.data_type()) {
-            (DataType::List(daft_child_dtype), arrow2::datatypes::DataType::List(arrow_child_field)) |
-            (DataType::List(daft_child_dtype), arrow2::datatypes::DataType::LargeList(arrow_child_field))
-            => {
-                let arrow_arr = arrow_arr.to_type(arrow2::datatypes::DataType::LargeList(arrow_child_field.clone()));
-                let arrow_arr = arrow_arr.as_any().downcast_ref::<arrow2::array::ListArray<i64>>().unwrap();
+            (
+                DataType::List(daft_child_dtype),
+                arrow2::datatypes::DataType::List(arrow_child_field),
+            )
+            | (
+                DataType::List(daft_child_dtype),
+                arrow2::datatypes::DataType::LargeList(arrow_child_field),
+            ) => {
+                let arrow_arr = arrow_arr.to_type(arrow2::datatypes::DataType::LargeList(
+                    arrow_child_field.clone(),
+                ));
+                let arrow_arr = arrow_arr
+                    .as_any()
+                    .downcast_ref::<arrow2::array::ListArray<i64>>()
+                    .unwrap();
                 let arrow_child_array = arrow_arr.values();
-                let child_series = Series::from_arrow(Arc::new(Field::new("list", daft_child_dtype.as_ref().clone())), arrow_child_array.clone())?;
+                let child_series = Series::from_arrow(
+                    Arc::new(Field::new("list", daft_child_dtype.as_ref().clone())),
+                    arrow_child_array.clone(),
+                )?;
                 Ok(ListArray::new(
                     field.clone(),
                     child_series,
                     arrow_arr.offsets().clone(),
                     arrow_arr.validity().cloned(),
                 ))
             }
-            (d, a) => Err(DaftError::TypeError(format!("Attempting to create Daft FixedSizeListArray with type {} from arrow array with type {:?}", d, a)))
+            (
+                DataType::List(daft_child_dtype),
+                arrow2::datatypes::DataType::Map(arrow_child_field, keys_sorted),
+            ) => {
+                let arrow_arr = arrow_arr.to_type(arrow2::datatypes::DataType::Map(
```

Comment:
Why is this line necessary? Doesn't line 102 already establish that the `arrow_arr` has a Map type?

Reply:
ah, nice catch!